src/arcaflow_plugin_sdk/test_schema.py

```diff
@@ -521,159 +534,82 @@ class OneOfData2:
-                    {
-                        "type": PropertyType(
-                            schema.StringType(),
-                        ),
-                        "a": PropertyType(schema.StringType()),
-                    },
-                ),
-                "b": schema.ObjectType(
-                    OneOfData2, {"b": PropertyType(schema.IntType())}
-                ),
+                "a": schema.RefType("a", self.scope_embedded),
+                "b": schema.RefType("b", self.scope_embedded),
             },
-            "a",
-        )
-        s = schema.OneOfStringType(
-            {"a": schema.RefType("a", scope), "b": schema.RefType("b", scope)},
-            scope,
-            "type",
+            scope=self.scope_embedded,
+            discriminator_field_name="type",
         )
 
-        unserialized_data: OneOfData1 = s.unserialize(
+        unserialized_data: OneOfTest.OneOfDataEmbedded1 = s.unserialize(
             {"type": "a", "a": "Hello world!"}
         )
-        self.assertIsInstance(unserialized_data, OneOfData1)
+        self.assertIsInstance(unserialized_data, OneOfTest.OneOfDataEmbedded1)
         self.assertEqual(unserialized_data.type, "a")
         self.assertEqual(unserialized_data.a, "Hello world!")
 
-        unserialized_data2: OneOfData2 = s.unserialize({"type": "b", "b": 42})
-        self.assertIsInstance(unserialized_data2, OneOfData2)
+        unserialized_data2: OneOfTest.OneOfData2 = s.unserialize(
+            {"type": "b", "b": 42}
+        )
+        self.assertIsInstance(unserialized_data2, OneOfTest.OneOfData2)
         self.assertEqual(unserialized_data2.b, 42)
 
     def test_validation(self):
-        @dataclasses.dataclass
-        class OneOfData1:
-            type: str
-            a: str
-
-        @dataclasses.dataclass
-        class OneOfData2:
-            b: int
-
-        scope = schema.ScopeType(
+        s = schema.OneOfStringType[OneOfTest.OneOfDataEmbedded1](
             {
-                "a": schema.ObjectType(
-                    OneOfData1,
-                    {
-                        "type": PropertyType(
-                            schema.StringType(),
-                        ),
-                        "a": PropertyType(schema.StringType()),
-                    },
-                ),
-                "b": schema.ObjectType(
-                    OneOfData2, {"b": PropertyType(schema.IntType())}
-                ),
+                "a": schema.RefType("a", self.scope_embedded),
+                "b": schema.RefType("b", self.scope_embedded),
             },
-            "a",
-        )
-        s = schema.OneOfStringType[OneOfData1](
-            {"a": schema.RefType("a", scope), "b": schema.RefType("b", scope)},
-            scope,
-            "type",
+            scope=self.scope_embedded,
+            discriminator_field_name="type",
         )
 
         with self.assertRaises(ConstraintException):
             # noinspection PyTypeChecker
```

Comment:
Do we still need this directive?  (Why don't the types match??)  Or, are we trying to prove that, given a type mismatch, we'll raise an exception?

Reply:
To me, it looks like we are trying to show that given a `None` we will raise an exception.

I recommend opening an IDE or executing PyTypeChecker to see what happens when you remove the directive.

Reply:
I could do that, but I would have to pull your branch and do all that stuff, whereas you've got it right in front of you.  😇

Reply:
I didn't add that test. You would just need `main`.